commands/release.js

```diff
@@ -86,16 +86,35 @@ function createCompileCache(CONFIG, done) {
  * @param {Function} done
  * @api public
  */
-function createBrandedApplication(CONFIG, done) {
+const createBrandedApplication = (CONFIG, done) => {
   cli.debug('running electron-packager');
   packager(CONFIG.packagerOptions, function(err, res) {
     if (err) {
       return done(err);
     }
     cli.debug('Packager result is: ' + JSON.stringify(res, null, 2));
-    done(null, true);
+    if (CONFIG.platform !== 'darwin') {
+      return done(null, true);
+    }
+
+    /**
+     * @see https://jira.mongodb.org/browse/INT-1836
+     */
+    cli.debug('Ensuring `Contents/MacOS/Electron` is symlinked');
+
+    const cwd = process.cwd();
+
+    process.chdir(path.join(CONFIG.appPath, 'Contents', 'MacOS'));
+
+    fs.ensureSymlink(CONFIG.productName, 'Electron', function(_err) {
```

Comment:
Yep there is a guard on `CONFIG.platform` so this will only be run on OS X.  Thanks for checking this out!